api/apps/api/src/modules/features/domain/feature-specification.spec.ts

```diff
@@ -0,0 +1,110 @@
+import {
+  FeatureSpecification,
+  SpecificationStatus,
+} from './feature-specification';
+import { FeatureSpecificationId } from './feature-specification.id';
+import { FeatureSpecificationRevision } from './feature-specification-revision';
+
+describe(`when specification was already created`, () => {
+  let spec: FeatureSpecification;
+  beforeEach(() => {
+    spec = new FeatureSpecification(
+      new FeatureSpecificationId(`spec-id`),
+      SpecificationStatus.Created,
+      [],
+      new FeatureSpecificationRevision(5),
+    );
+  });
+
+  describe(`when submitting again`, () => {
+    beforeEach(() => {
+      spec.submit();
+    });
+
+    it(`should emit compute event`, () => {
+      expect(spec.getUncommittedEvents()).toMatchInlineSnapshot(`
+        Array [
+          ComputeFeatureSpecificationEvent {
+            "featureSpecificationId": "spec-id",
+            "revision": 6,
+          },
+        ]
+      `);
+    });
+
+    it(`should change state`, () => {
+      expect(spec.getStatus()).toEqual(SpecificationStatus.Created);
+    });
+    it(`should set revision to 6`, () => {
+      expect(spec.getRevision()).toEqual(6);
+    });
+  });
+
+  describe(`when changing spec`, () => {
+    beforeEach(() => {
+      spec.change({
+        features: [],
+      });
+    });
+
+    it(`should not emit compute event`, () => {
+      expect(spec.getUncommittedEvents()).toEqual([]);
+    });
+    it(`should put into draft state`, () => {
+      expect(spec.getStatus()).toEqual(SpecificationStatus.Draft);
+    });
+    it(`should generate new revision number`, () => {
+      expect(spec.getRevision()).toEqual(6);
+    });
+  });
+});
+
+describe(`when creating brand new specification`, () => {
+  let spec: FeatureSpecification;
+  beforeEach(() => {
+    spec = new FeatureSpecification(
+      new FeatureSpecificationId(`spec-id`),
+      SpecificationStatus.Draft,
+      [],
+    );
```

Comment:
this is ok but let's not forget that an initial status of `Draft` is not going to be necessarily the case - `Created` can be set straight away (the documentation uses `Draft` just for simplicity's sake)

Reply:
https://github.com/Vizzuality/marxan-cloud/pull/369#discussion_r676462727
Feel like the very same case.